### PR TITLE
fix(contrib): check is runresult is part of a release

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#21420.0 ✘  
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✘  
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-cds#21421.0 ✘  
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test UI ✔

File: contrib/grpcplugins/action/artifactory-plugins-lib/artifactory-plugins-lib.go [hunk 1]
```diff
@@ -133,7 +140,28 @@ func promoteRunResult(ctx context.Context, c *actionplugin.Common, artifactClien
...
+		}
+		releaseValues, has := props["release.name"]
+		if has && len(releaseValues) > 0 {
+			if releaseValues[0] != futureReleaseName {
+				return promotedArtifact, errors.Errorf("run result %s is already part of another release: %s ", promotedArtifact, releaseValues[0])
+			}
+			releaseVersion, has := props["release.version"]
+			if has && len(releaseVersion) > 0 {
+				if releaseVersion[0] != futureReleaseVersion {
+					return promotedArtifact, errors.Errorf("run result %s has already been release in version %s", promotedArtifact, releaseVersion[0])
```
Comment (fsamin) [inline contrib/grpcplugins/action/artifactory-plugins-lib/artifactory-plugins-lib.go:160]:
```suggestion
					return promotedArtifact, errors.Errorf("Run result %s has already been release in version %s", promotedArtifact, releaseVersion[0])
```

Comment (fsamin) [reply contrib/grpcplugins/action/artifactory-plugins-lib/artifactory-plugins-lib.go:160]: We should write error with grpcplugins.Errorf to have a red message in the job output

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#21421.1 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-cds#21423.0 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21423.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

**Review by richardlt — APPROVED** [review]

**Pull request merged**